Problem2.c

```diff
@@ -0,0 +1,79 @@
+/* Problem Statement:
+Question not present on LeetCode
+
+Given an array of numbers of length N, find both the minimum and maximum. Follow up : Can you do it using less than 2 * (N - 2) comparison
+
+ *
+ * Time Complexity : O(n) specifically 2 * O(N - 2)
+ * STILL TRYING TO FIGURE OUT THE CASE WHERE WE CAN SOLVE BETTER THAN 2 * O(N - 2)
+ * Space Complexity : O(1) 
+ */
+
+#include<stdio.h>
+#include<stdlib.h>
+#include<string.h>
+#include<stdbool.h>
+#include<math.h>
+
+
+/**
+ * Note: The returned array must be malloced, assume caller calls free().
+ */
+void find_min_max(int* nums, int numsSize){
+    int idx = 0, min = 0, max = 0;
+    
+    /* validation checks */
+    if (!nums) {
+        return;
+    }
+
+    /* Minimum 2 numbers needed */
+    if (numsSize < 2 ){
+        printf("Insufficient data to perform operations \n");
+        return;
+    }
+    /* Initialize min and max */
+    min = nums[0];
+    max = nums[1];
+
+    /* Go over the numbers from 3rd element onwards and based on condition update min and max.*/
+    for (idx = 2; idx < numsSize; idx++) {
+        if (min > nums[idx]) {
+            min = nums[idx];
+        } else if (max < nums[idx]) {
```

Comment:
why have you added else if condition?
is else required?

Reply:
Hi Raman, I have added else if condition as I was not able to find any case where a min is updated and then max will also get updated in the same condition. Let me know if I am wrong here.
Also I just realised that I set min max for the first 2 elements without even comparing them. I will fix it too.